regression/test.pl

```diff
@@ -33,7 +33,12 @@ sub with_color {
 
 sub run($$$$$) {
   my ($name, $input, $cmd, $options, $output) = @_;
-  my $cmdline = "$cmd $options '$input' >'$output' 2>&1";
+  my $cmdline;
+  if(length($input)) {
+    $cmdline = "$cmd $options '$input' >'$output' 2>&1";
+  } else {
+    $cmdline = "$cmd $options >'$output' 2>&1";
+  }
```

Comment:
Does this actually address #3981?

Reply:
Yes, it does.

Reply:
Cool! Please add the "Fixes: #3981" info to the commit message to have it auto-closed.